package.json

```diff
@@ -51,6 +60,8 @@
     "reboot:full": "npm run clean:full",
     "release": "node scripts/release",
     "reset": "npx lerna run --stream reset",
+    "reset:full": "npx npm-run-all reset reset:top",
+    "reset:top": "npx rimraf .nyc_output coverage",
```

Comment:
Why `reset:top`? Could this be `reset:coverage` instead?

Reply:
We've not needed to "reset" the top in the past, though we've had a `clean:top` for some time. In hindsight, I probably should have used `[script]:root` naming as that might communicate the purpose more clearly — e.g. to `reset` and `clean` the root of the monorepo apart from member packages.

In the member packages, pkgJson `reset` does things like delete `build` artifacts, similar to the behavior of `embark reset` in dapps. pkgJson `clean`'s purpose is to put the package back to an uninitialized state and each package's `clean` script calls that package's `reset` script before doing whatever else it does. At present, `clean` doesn't do much extra in the member packages because we rely on `lerna clean` from the root to delete their `node_modules` directories, but more actions can be added as needed to individual `clean` scripts.

With that in mind, the changes I made put the same logic in place at the root (the top) — do reset stuff, then do clean stuff. I've kept it generalized so that if we need to reset other root stuff in the future the list of files/dirs passed to rimraf can simply be expanded.